src/mainwindow.cpp

```diff
@@ -1686,66 +1686,91 @@ void MainWindow::activateSelectedProfile()
 
-    if (ui->profileBox->currentIndex() == 0) {
-      ui->profileBox->setCurrentIndex(previousIndex);
+    // show the dialog
+    ProfilesDialog dlg(previousName, m_OrganizerCore, this);
+    dlg.exec();
 
-      std::optional<QString> newSelection;
+    // check if the user clicked 'select' to select another profile
+    std::optional<QString> newSelection = dlg.selectedProfile();
 
-      ProfilesDialog dlg(ui->profileBox->currentText(), m_OrganizerCore, this);
+    // refresh the profile box; this loops until there is at least one profile
+    // available, which shouldn't really happen because the dialog won't allow
+    // it
+    //
+    // the `false` to refreshProfiles() is so it doesn't try to select the
+    // profile in the list because 1) it's done just below, and 2) it might be
+    // wrong profile if there's something in newSelection
+    while (!refreshProfiles(false)) {
+      ProfilesDialog dlg(previousName, m_OrganizerCore, this);
       dlg.exec();
       newSelection = dlg.selectedProfile();
+    }
 
-      while (!refreshProfiles()) {
-        ProfilesDialog dlg(ui->profileBox->currentText(), m_OrganizerCore, this);
-        dlg.exec();
-        newSelection = dlg.selectedProfile();
-      }
-
-      if (newSelection) {
-        ui->profileBox->setCurrentText(*newSelection);
-        activateSelectedProfile();
-      }
+    // note that setCurrentText() is recursive, it will re-execute this function
+    if (newSelection) {
+      ui->profileBox->setCurrentText(*newSelection);
     } else {
-      activateSelectedProfile();
+      ui->profileBox->setCurrentText(previousName);
     }
 
-    LocalSavegames *saveGames = m_OrganizerCore.managedGame()->feature<LocalSavegames>();
-    if (saveGames != nullptr) {
-      if (saveGames->prepareProfile(m_OrganizerCore.currentProfile())) {
-        m_SavesTab->refreshSaveList();
-      }
+    // nothing else to do because setCurrentText() is recursive and will
+    // have re-executed on_profileBox_currentIndexChanged() again, doing all
+    // the stuff below for the new selection
+    return;
+  }
+
+
+  activateSelectedProfile();
+
+  LocalSavegames *saveGames = m_OrganizerCore.managedGame()->feature<LocalSavegames>();
+  if (saveGames != nullptr) {
+    if (saveGames->prepareProfile(m_OrganizerCore.currentProfile())) {
+      m_SavesTab->refreshSaveList();
     }
+  }
 
-    BSAInvalidation *invalidation = m_OrganizerCore.managedGame()->feature<BSAInvalidation>();
-    if (invalidation != nullptr) {
-      if (invalidation->prepareProfile(m_OrganizerCore.currentProfile())) {
-        QTimer::singleShot(5, &m_OrganizerCore, SLOT(profileRefresh()));
-      }
+  BSAInvalidation *invalidation = m_OrganizerCore.managedGame()->feature<BSAInvalidation>();
+  if (invalidation != nullptr) {
+    if (invalidation->prepareProfile(m_OrganizerCore.currentProfile())) {
+      QTimer::singleShot(5, &m_OrganizerCore, SLOT(profileRefresh()));
```

Comment:
Can this be changed to not use the old `SLOT()` syntax?

Reply:
One day, maybe.

Reply:
My test instance doesn't get to that line and it works fine, so not for this PR.